examples/shader/post_processing.rs

```diff
@@ -114,59 +120,41 @@ impl Plugin for PostProcessPlugin {
 }
 
 /// The post process node used for the render graph
-struct PostProcessNode {
-    // The node needs a query to gather data from the ECS in order to do its rendering,
-    // but it's not a normal system so we need to define it manually.
-    query: QueryState<&'static ViewTarget, With<ExtractedView>>,
-}
-
+#[derive(Default)]
+struct PostProcessNode;
 impl PostProcessNode {
     pub const NAME: &str = "post_process";
 }
 
-impl FromWorld for PostProcessNode {
-    fn from_world(world: &mut World) -> Self {
-        Self {
-            query: QueryState::new(world),
-        }
-    }
-}
-
-impl Node for PostProcessNode {
-    // This will run every frame before the run() method
-    // The important difference is that `self` is `mut` here
-    fn update(&mut self, world: &mut World) {
-        // Since this is not a system we need to update the query manually.
-        // This is mostly boilerplate. There are plans to remove this in the future.
-        // For now, you can just copy it.
-        self.query.update_archetypes(world);
-    }
+// The ViewNode trait is required by the ViewNodeRunner
+impl ViewNode for PostProcessNode {
+    // The node needs a query to gather data from the ECS in order to do its rendering,
+    // but it's not a normal system so we need to define it manually.
+    //
+    // This query will only run on the view entity
+    type ViewQuery = &'static ViewTarget;
 
     // Runs the node logic
     // This is where you encode draw commands.
     //
-    // This will run on every view on which the graph is running. If you don't want your effect to run on every camera,
-    // you'll need to make sure you have a marker component to identify which camera(s) should run the effect.
+    // This will run on every view on which the graph is running.
+    // If you don't want your effect to run on every camera,
+    // you'll need to make sure you have a marker component
```

Comment:
```suggestion
    // you'll need to make sure you have a marker component as part of ViewQuery
```
(link to the type with rustdoc)